### DISCUSSION TITLE
SVG support for own website?

### DISCUSSION THREAD

**Discussion by UweKeim** [darkreader/darkreader#7577] [Q&A] [answered]
I am using the [Dark Reader JS script for my own website](https://github.com/darkreader/darkreader#using-dark-reader-for-a-website). 

I have several monochrome SVG icons on my page.

This is how it looks like in light mode:

![image](https://user-images.githubusercontent.com/415439/145076991-aab67dec-4b59-4d27-8e41-e428c11d5164.png)

And this is how it looks like in dark mode via Dark Reader:

![image](https://user-images.githubusercontent.com/415439/145077142-f9241a85-5ff6-46ec-a930-81ff4a9aab40.png)

([This is the page](https://www.zeta-software.de/index.html), in case it matters).

As you can see, the SVG icon is not being modified by Dark Reader.

Ideally this should look like e.g.:

![image](https://user-images.githubusercontent.com/415439/145077585-ce285fe9-de5b-4934-bdb5-0ceead1e7f5a.png)

------

I was looking through [the CDN source file](https://unpkg.com/browse/darkreader@4.9.40/darkreader.js) and found several occurrences of the word "svg".

So I do _think_ that Dark Reader somehow can handle SVG images, but I did not find any further documentation about whether/how I can activate/configure SVG support when including Dark Reader on my own website.

I am thinking about additional configuration options when calling the Dark Reader JS methods or giving my SVG image HTML tags some addition HTML attributes or something like that. Again, I did not find any examples or documentation on that idea.

### My question

Is there any way to tell Dark Reader to colorize some/all SVGs on a web page?

**Comment by Gusted** [comment, marked as answer]
> Is there any way to tell Dark Reader to colorize some/all SVGs on a web page?

Yeah, that's possible. Given your site, you have to do a small change.

```typescript
DarkReader.enable({... options here ...});
```
->
```typescript
DarkReader.enable({... options here ...}, {invert: ['img[src^=".svg"]']});
```

**Comment by donfiguerres** [comment]
Hi everyone! It's not working for me when I'm trying to use it on my website. Below is my code.

```
  DarkReader.enable({
      brightness: 100,
      contrast: 100,
      sepia: 0
  }, {invert: ['img[src^=".svg"]']});
```

But the SVG images are not inverted.
![image](https://user-images.githubusercontent.com/12525903/172987474-57b1b0f8-d5f5-41b3-8f73-4a41521f9d17.png)

Do you have a link to the documentation or example on how to exactly do this? Thanks

**Reply by UweKeim** [reply]
Depending on your URLs, it might be necessary to use `*` instead of `^` or `$`:

```js
DarkReader.enable({
      brightness: 100,
      contrast: 100,
      sepia: 0
}, {invert: ['img[src*=".svg"]']});
```

See [CSS Selector examples](https://www.w3schools.com/cssref/css_selectors.asp).

In my projects, I often let the framework add a version string to the URL like e.g.:

```html
<img src="/logo.svg?v=uV2njFkouQCCuWYW_MhzJJv99hilUuxk9ojDlevkmh4" />
```

Therefore, the [`*` ("contain")](https://www.w3schools.com/cssref/sel_attr_contain.asp) is required.

---------

On a more general note, I do think the [`^` ("begin")](https://www.w3schools.com/cssref/sel_attr_begin.asp) is always wrong. Instead you usually want the [`$` ("end")](https://www.w3schools.com/cssref/sel_attr_end.asp) instead:

```js
DarkReader.enable({
      brightness: 100,
      contrast: 100,
      sepia: 0
}, {invert: ['img[src$=".svg"]']});
```

**Reply by donfiguerres** [reply]
That worked. Thanks!
